source/common/grpc/async_client_manager_impl.h

```diff
@@ -55,13 +55,46 @@ class AsyncClientManagerImpl : public AsyncClientManager {
                                               Stats::Scope& scope,
                                               bool skip_cluster_check) override;
 
+  RawAsyncClientSharedPtr
+  getOrCreateRawAsyncClient(const envoy::config::core::v3::GrpcService& config, Stats::Scope& scope,
+                            bool skip_cluster_check) override {
+    RawAsyncClientSharedPtr client;
+    client = thread_local_cache_->getCache(config);
+    if (client != nullptr) {
+      return client;
+    }
+    client = factoryForGrpcService(config, scope, skip_cluster_check)->create();
+    thread_local_cache_->setCache(config, client);
+    return client;
+  }
+
 private:
+  class ThreadLocalCache : public ThreadLocal::ThreadLocalObject {
```

Comment:
A nit: Could you use a less generic name here? Perhaps something that mentions grpc client?

Reply:
Thanks, makes sense.